code/modules/recycling/sortingmachinery.dm

```diff
@@ -272,13 +268,11 @@
 		return
 
 	if(action == "select_destination")
-		for(var/destination in GLOB.TAGGERLOCATIONS)
-			var/destination_id = text2num(params["destination"])
-			if(currTag != destination_id && length(GLOB.TAGGERLOCATIONS) >= destination_id)
-				currTag = destination_id
-				playsound(src, 'sound/machines/terminal_select.ogg', 15, TRUE)
-				break
-		add_fingerprint(usr)
+		var/destination_id = text2num(params["destination"])
+		if(currTag != destination_id && length(GLOB.TAGGERLOCATIONS) >= destination_id)
```

Comment:
> `length(GLOB.TAGGERLOCATIONS) >= destination_id`

This should never happen outside of malicious client input.
However, if we're guarding against malicious clients, then we should also consider them supplying negative `destination_id`, which would result in exact same sort of runtimes.

So i think this check needs to either be completely removed
or supplemented to look something like `1 <= destination_id && destination_id <= length(GLOB.TAGGERLOCATIONS)`, or, fancier `destination_id = clamp(destination_id, 1, length(GLOB.TAGGERLOCATIONS))`
Or, move it from the if towards the line above
```dm
var/destination_id = clamp(text2num(params["destination"]), 1, length(length(GLOB.TAGGERLOCATIONS))`
```

(i don't have an opinion on which way of these is better; not having this check at all is fine by me as well)